stl/inc/list

```diff
@@ -1806,43 +1806,52 @@ void swap(list<_Ty, _Alloc>& _Left, list<_Ty, _Alloc>& _Right) noexcept /* stren
 }
 
 template <class _Ty, class _Alloc>
-_NODISCARD bool operator==(const list<_Ty, _Alloc>& _Left, const list<_Ty, _Alloc>& _Right) {
+_NODISCARD bool operator==(const list<_Ty, _Alloc>& _Left, const list<_Ty, _Alloc>& _Right) noexcept(
+    noexcept(_STD equal(_Left.begin(), _Left.end(), _Right.begin()))) {

```

Comment:
`equal` is never marked as `noexcept` so this will always be false. Ditto `lexicographical_compare(_three_way)` below.
https://github.com/microsoft/STL/blob/3cafa97eecdbfde41ea5c09126f877a7eb97f9e9/stl/inc/xutility#L4825-L4829
https://github.com/microsoft/STL/blob/3cafa97eecdbfde41ea5c09126f877a7eb97f9e9/stl/inc/xutility#L5148-L5152
https://github.com/microsoft/STL/blob/3cafa97eecdbfde41ea5c09126f877a7eb97f9e9/stl/inc/xutility#L5178-L5180

Reply:
The lexographical compare was likely a merge that I missed. I'm porting these from an earlier set of headers and there were definitely some areas I had to merge with changes for spaceship.

I'll need to double check what happened with equal.